src/services/validationService.ts

```diff
@@ -0,0 +1,155 @@
+  }
+
+  private static validateDateRange(
+    dateRange: string[] | string = "",
+    allowedRange: Number = 0
+  ): IValidationResponse {
+    if (dateRange) {
+      const extracted = Array.isArray(dateRange)
+        ? dateRange[0].split("/")
+        : dateRange.toString().split("/");
+      const fromDate = new Date(extracted[0]).getTime();
+      const toDate = new Date(extracted[1]).getTime();
+      const diffInDays = (toDate - fromDate) / (1000 * 3600 * 24);
+      if (diffInDays > allowedRange) {
+        return {
+          errorMessage: constants.invalidDateRange.replace(
+            "${allowedRange}",
+            allowedRange.toString()
+          ),
+          isValid: false,
+        };
+      } else {
+        return { isValid: true };
+      }
+    } else {
+      return {
+        errorMessage: constants.noDataRange,
+        isValid: false,
+      };
+    }
+  }
+
+  private static validateCommonRules(query: IQuery, limits: ICommon): void {
+    if (query.threshold) {
+      query.threshold =
+        query.threshold > limits.max_result_threshold
+          ? limits.max_result_threshold
+          : query.threshold || limits.max_result_threshold;
+    } else {
+      query.threshold = limits.max_result_threshold;
+    }
+    if (query.limit) {
+      query.limit =
+        query.limit > limits.max_result_row_limit
+          ? limits.max_result_row_limit
+          : query.limit || limits.max_result_row_limit;
+    } else {
+      query.limit = limits.max_result_row_limit;
+    }
+  }
+  public static validateSqlQuery(
+    queryData: IString,
+    limits: ILimits
+  ): IValidationResponse {
+    if (limits != undefined && limits) {
+      const isDateMentioned = queryData.query.indexOf("TIMESTAMP");
+      this.validateSqlLimits(queryData, limits.common.max_result_row_limit);
+      const allowedRange = limits.queryRules.scan.max_date_range || 0;
+      if (isDateMentioned != -1) {
+        let fromDate: any = new Date(
+          queryData.query
+            .substring(queryData.query.indexOf("TIMESTAMP"))
+            .split(" ")[1]
+        ).getTime();
+        let toDate: any = new Date(
+          queryData.query
+            .substring(queryData.query.lastIndexOf("TIMESTAMP"))
+            .split(" ")[1]
+        ).getTime();
+        const diffInDays = (toDate - fromDate) / (1000 * 3600 * 24);
+        if (diffInDays > allowedRange || diffInDays < 0) {
+          return {
+            errorMessage: constants.invalidDateRange.replace(
+              "${allowedRange}",
+              allowedRange.toString()
+            ),
+            isValid: false,
+          };
+        } else {
+          return { isValid: true };
+        }
+      } else {
+        return {
+          errorMessage: constants.noDataRange,
+          isValid: false,
+        };
+      }
+    } else {
+      return { isValid: true };
+    }
+  }
+  private static validateSqlLimits(
+    queryData: IString,
+    maxLimit: Number = 0
+  ): void {
```

Comment:
don't implement the same logic for the SQL as well.  create the generic logic and use them in both SQL and native query validation.